### PR TITLE
[stable/prometheus-operator] kubelet servicemonitor skip tls verify

### PR DESCRIPTION
**Pull request by via-jordan-sokolic**
No

#### What this PR does / why we need it:

Addresses issue introduced in #22586 - scrape config is added without `tlsConfig.insecureSkipVerify: true` which results in Prometheus complaining about CA verification:
> Get "https://xx.xx.xx.xx:10250/metrics/resource/v1alpha1": x509: certificate signed by unknown authority 

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #22586 

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

**Comment by den-is** [comment]
that is already addressed in https://github.com/helm/charts/pull/22918

**Comment by via-jordan-sokolic** [comment]
> that is already addressed in #22918

Oh, looks like @tklovett beat me to it by a couple of hours :)
In this PR I also include the `kubelet.serviceMonitor.interval` plus a minor modification to template syntax (missing newline chomp), otherwise it's indeed the same as #22918 .

**Comment by tklovett** [comment]
> In this PR I also include the kubelet.serviceMonitor.interval plus a minor modification to template syntax (missing newline chomp), otherwise it's indeed the same as #22918 .

These changes are probably good to have too 👍 I don't mind if we want to close mine in favor of doing all these changes in one go.

Also, should there be an additional non-TLS `http-metrics` version, like there is for `path: /metrics/cadvisor`?

**Comment by via-jordan-sokolic** [comment]
> Also, should there be an additional non-TLS `http-metrics` version, like there is for `path: /metrics/cadvisor`?

Hm, yes, probably a good idea to add that.

File: stable/prometheus-operator/templates/exporters/kubelet/servicemonitor.yaml [hunk 1]
```diff
@@ -6,7 +6,7 @@ metadata:
   namespace: {{ template "prometheus-operator.namespace" . }}
   labels:
     app: {{ template "prometheus-operator.name" . }}-kubelet
-{{ include "prometheus-operator.labels" . | indent 4 }}
+{{- include "prometheus-operator.labels" . | indent 4 }}
```
Comment (tklovett) [inline stable/prometheus-operator/templates/exporters/kubelet/servicemonitor.yaml:9]: I believe you need the newline after `}}-kubelet` above, but this `{{-` whitespace chomper will remove it.

```suggestion
{{ include "prometheus-operator.labels" . | indent 4 }}
```

Comment (via-jordan-sokolic) [reply stable/prometheus-operator/templates/exporters/kubelet/servicemonitor.yaml:9]: @tklovett I get this output with the newline chomp:

    $ helm template prometheus-operator --namespace monitoring ./stable/prometheus-operator
```yaml
# Source: prometheus-operator/templates/exporters/kubelet/servicemonitor.yaml
apiVersion: monitoring.coreos.com/v1
kind: ServiceMonitor
metadata:
  name: prometheus-operator-kubelet
  namespace: monitoring
  labels:
    app: prometheus-operator-kubelet    
    chart: prometheus-operator-8.15.7
    release: "prometheus-operator"
    heritage: "Helm"
spec:
  endpoints:
```

And this without:
```yaml
# Source: prometheus-operator/templates/exporters/kubelet/servicemonitor.yaml
apiVersion: monitoring.coreos.com/v1
kind: ServiceMonitor
metadata:
  name: prometheus-operator-kubelet
  namespace: monitoring
  labels:
    app: prometheus-operator-kubelet
    
    chart: prometheus-operator-8.15.7
    release: "prometheus-operator"
    heritage: "Helm"
spec:
  endpoints:
```

Not sure why you think it's necessary?

Comment (tklovett) [reply stable/prometheus-operator/templates/exporters/kubelet/servicemonitor.yaml:9]: Ah, my bad. I didn't realize there was also a newline coming in from the [`prometheus-operator.labels`](https://github.com/helm/charts/blob/05966cd7f3e712ce82aec1c884d913b9c62ed704/stable/prometheus-operator/templates/_helpers.tpl#L48-L49) template too. You have it right 👍

Comment (via-jordan-sokolic) [reply stable/prometheus-operator/templates/exporters/kubelet/servicemonitor.yaml:9]: Cool, marking as resolved.

**Comment by vsliouniaev** [comment]
Looks like you need to bump the chart version again
/ok-to-test
/lgtm

**Comment by via-jordan-sokolic** [comment]
Resolving merge conflict through the Github kind of borked the branch -- hoping the force update didn't bork it further..

**Pull request merged**